client/src/hooks/useSignup.js

```diff
@@ -1,41 +1,47 @@
-import { useState } from 'react'
-import { useNavigate } from 'react-router-dom';
-import { useAuthContext } from './useAuthContext'
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useCookies } from "react-cookie";
 
 export const useSignup = () => {
-    const navigate =  useNavigate();
-  const [error, setError] = useState(null)
-  const [isLoading, setIsLoading] = useState(null)
-  const { dispatch } = useAuthContext()
-
-  const signup = async (email, password) => {
-    setIsLoading(true)
-    setError(null)
-
-    const response = await fetch('/api/user/signup', {
-      method: 'POST',
-      headers: {'Content-Type': 'application/json'},
-      body: JSON.stringify({ email, password })
-    })
-    const json = await response.json()
+  const navigate = useNavigate();
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(null);
+  const [cookies, setCookie, removeCookie] = useCookies(null);
+
+  
+
+  const signup = async (name, email, password) => {
+    setIsLoading(true);
+    setError(null);
+
+    const response = await fetch("/api/user/signup", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ name, email, password }),
+    });
+    const user = await response.json();
 
     if (!response.ok) {
-      setIsLoading(false)
-      setError(json.error)
+      setIsLoading(false);
+      setError(user.error);
     }
     if (response.ok) {
       // save the user to local storage
-      localStorage.setItem('user', JSON.stringify(json))
+      const localUser = JSON.stringify(user);
+      localStorage.setItem("user", localUser);
```

Comment:
Should you use local storage to store sensitive user details ?

1. https://www.horangi.com/blog/misuse-of-local-storage#:~:text=Local%20storage%20should%20only%20be,Physical%20access%20to%20the%20browser
2. https://snyk.io/blog/is-localstorage-safe-to-use/

Checkout above given links, to identify.